utils/helpers.ts

```
@@ -107,39 +146,96 @@ export function formatDuration(seconds: number): string {
-  const networkDiff = (BigInt(accepted) / BigInt(Math.round(Number(difficulty) * 100))) * BigInt(10000);
+/**
+ * Calculates the probability of finding a block within various time periods.
+ * Returns early with minimal probabilities if inputs are zero.
+ * @param {bigint} hashRate - The hashrate in H/s
+ * @param {number} difficulty - The difficulty as a percentage of network difficulty
+ * @param {bigint} accepted - The number of accepted shares
+ * @returns {object} An object with keys '1h', '1d', '1w', '1m', '1y' mapping to probability strings (e.g., "0.123%", "<0.001%")
+ */
+export function calculateBlockChances(hashRate: bigint, difficulty: number, accepted: bigint): { [key: string]: string} {
+  const difficultyFactor = Math.round(Number(difficulty) * 100);
+  const acceptedBigInt = typeof accepted === 'bigint' ? accepted : BigInt(accepted);
```

ESLint

[error] Insert `⏎···`

(prettier/prettier)